src/paperbot/application/services/paper_search_service.py

```
@@ -0,0 +1,133 @@
+        duplicates_removed = total_raw - len(unique)
+
+        # 3. Persist if requested
+        if persist and self._registry:
+            for paper in unique:
+                try:
+                    result_dict = self._registry.upsert_paper(
+                        paper=paper.to_dict(),
+                        source_hint=provenance.get(paper.title_hash, ["unknown"])[0],
+                    )
+                    paper.canonical_id = result_dict.get("id")
+                except Exception as e:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)